saml-lib/src/test/java/uk/gov/ida/saml/security/signature/OutgoingKeySignatureTrustEngineTest.java

```diff
@@ -0,0 +1,108 @@
+        trustBasisCriteria.add(
+                new EntityIdCriterion(TestEntityIds.HUB_ENTITY_ID)
+        );
+        
+        Counter outgoingSignatureVerifyingErrorCounter = mock(Counter.class);
+        setFinalStatic(OutgoingKeySignatureTrustEngine.class.getDeclaredField("outgoingSignatureVerifyingErrorCounter"), outgoingSignatureVerifyingErrorCounter);
+
+        final Credential incomingSigningCredential = new TestCredentialFactory(TestCertificateStrings.HUB_TEST_PUBLIC_SIGNING_CERT, TestCertificateStrings.HUB_TEST_PRIVATE_SIGNING_KEY).getSigningCredential();
+        final Assertion assertion = AssertionBuilder.anAssertion().withSignature(SignatureBuilder.aSignature().withSigningCredential(incomingSigningCredential).build()).build();
+
+        assertThat(trustEngine.doValidate(assertion.getSignature(), trustBasisCriteria)).isEqualTo(true);
+        verifyNoInteractions(outgoingSignatureVerifyingErrorCounter);
+    }
+    
+    @Test
+    public void shouldSendErrorCounterValidatingWithOutgoingSigningCertificate() throws Exception {
+        final Credential incomingSigningCredential = new TestCredentialFactory(
+                TestCertificateStrings.EXPIRED_SIGNING_PUBLIC_CERT,
+                TestCertificateStrings.EXPIRED_SIGNING_PRIVATE_KEY
+        ).getSigningCredential();
+        final Credential outgoingSigningCredential = new TestCredentialFactory(
+                TestCertificateStrings.HUB_TEST_PUBLIC_SIGNING_CERT,
+                TestCertificateStrings.HUB_TEST_PRIVATE_SIGNING_KEY
+        ).getSigningCredential();
+        final CredentialResolver credentialResolver = new StaticCredentialResolver(
+                Arrays.asList(incomingSigningCredential, outgoingSigningCredential)
+        );
+        final KeyInfoCredentialResolver keyInfoResolver = DefaultSecurityConfigurationBootstrap.buildBasicInlineKeyInfoCredentialResolver();
+
+        OutgoingKeySignatureTrustEngine trustEngine = new OutgoingKeySignatureTrustEngine(credentialResolver, keyInfoResolver);
+
+        CriteriaSet trustBasisCriteria = new CriteriaSet();
+        trustBasisCriteria.add(
+                new EntityIdCriterion(TestEntityIds.HUB_ENTITY_ID)
+        );
+        
+        Counter outgoingSignatureVerifyingErrorCounter = mock(Counter.class);
+        Counter.Child childCounter = mock(Counter.Child.class);
+        setFinalStatic(OutgoingKeySignatureTrustEngine.class.getDeclaredField("outgoingSignatureVerifyingErrorCounter"), outgoingSignatureVerifyingErrorCounter);
+
+        final Assertion outgoing_assertion = AssertionBuilder.anAssertion().withSignature(SignatureBuilder.aSignature().withSigningCredential(outgoingSigningCredential).build()).build();
+        
+        when(outgoingSignatureVerifyingErrorCounter.labels(anyString())).thenReturn(childCounter);
+        doNothing().when(childCounter).inc();
+        
+        trustEngine.doValidate(outgoing_assertion.getSignature(), trustBasisCriteria);
```

Comment:
```suggestion
        assertThat(trustEngine.doValidate(outgoing_assertion.getSignature(), trustBasisCriteria)).isEqualTo(true);
```

Might be good to assert this, to show this Signature Trust Engine still validates when there's a bad credential